### PR TITLE
✏️ Added work item tag cmdlets to Get, Remove and Update, also fixed typos and casing

### LINKED ISSUE #419: No cmdlets to manage work item tags?
Could it be the case that are currently no cmdlets to manage [work item tags](https://docs.microsoft.com/en-us/rest/api/azure/devops/wit/tags?view=azure-devops-rest-6.0)?

I've noticed the `Add/Get/Remove-VSTeamBuildTag` cmdlets, which are _build_ tags, not _work item_ tags.

I feel like this could be a useful addition. Can I go ahead and launch a PR for this one? 🤔

### PR DESCRIPTION
**Pull request by smholvoet**
# PR Summary

Fixed casing and a couple of minor typos.

_Update by Maintainer (@SebastianSchuetze): New cmdlets where added by @smholvoet_

## Suggestion

Another suggestion (which I have not yet pushed to my branch) is reworking the **Notes** section. I personally think it's very annoying seeing relevant cmdlets mentioned there, which are not clickable. Granted, they are mentioned a bit further down in the **Related links** section, but I think making the cmdlets clickable in the **Notes** section prevents a user from having to scroll further down and find the exact cmdlet he was looking for in a list of (sometimes) _many_ more cmdlets:

image

I've noticed some cmdlets have related links which get added below the cmdlets which are displayed through the [prerequisites.md ](https://github.com/MethodsAndPractices/vsteam/blob/trunk/.docs/common/prerequisites.md)file, for example [Update-VSTeamPolicy](https://github.com/MethodsAndPractices/vsteam/blob/a5d2ebebe8277861904c71229c31ddd6a02f7fe9/.docs/Update-VSTeamPolicy.md), which I think is very useful.

In summary:

- Remove `<!-- #include "./common/related.md" -->` from the Related links section
- Rework the Notes section to include link, making things easier for users

Let me know what you think and I'll push additional changes to this branch should you agree with the above suggestion.

## PR Checklist

- [ ] [Write Help](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-help)
- [ ] [Write Unit Test](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#write-unit-test)
- [ ] [Update CHANGELOG.md](https://github.com/DarqueWarrior/vsteam/blob/master/.github/CONTRIBUTING.md#update-changelogmd)

### CODE REVIEW

**Comment by SebastianSchuetze** [comment]
> Another suggestion (which I have not yet pushed to my branch) is reworking the **Notes** section. I personally think it's very annoying seeing relevant cmdlets mentioned there, which are not clickable. Granted, they are mentioned a bit further down in the **Related links** section, but I think making the cmdlets clickable in the **Notes** section prevents a user from having to scroll further down and find the exact cmdlet he was looking for in a list of (sometimes) _many_ more cmdlets:
> 
> image

@smholvoet Are you talking about the github docs pages regarding the links of the source markdown pages?

**Comment by smholvoet** [comment]
@SebastianSchuetze I'm indeed talking about the docs page, for example: [Get-VSTeamAccounts](https://methodsandpractices.github.io/vsteam-docs/docs/modules/vsteam/commands/Get-VSTeamAccounts#notes).

So in short: 

1. Remove the fixed commands which get added to the _Related links_ section, which has been added to every command:
image

2. Rework the [_Notes_ section](https://github.com/MethodsAndPractices/vsteam/blob/trunk/.docs/common/prerequisites.md) to include direct links to any cmdlet that gets mentioned there.

**Comment by SebastianSchuetze** [comment]
@DarqueWarrior what was the intention to add the standard related texts in every cmdlet? Is this nessecary to mention it everywhere?

**Comment by SebastianSchuetze** [comment]
@smholvoet  you can remove the related part. You are right, that it may be deserve it's own page once instead of having it on every cmdlet.

> @SebastianSchuetze I'm indeed talking about the docs page, for example: [Get-VSTeamAccounts](https://methodsandpractices.github.io/vsteam-docs/docs/modules/vsteam/commands/Get-VSTeamAccounts#notes).
> 
> So in short:
> 
> 1. Remove the fixed commands which get added to the _Related links_ section, which has been added to every command:
>    image
> 2. Rework the [_Notes_ section](https://github.com/MethodsAndPractices/vsteam/blob/trunk/.docs/common/prerequisites.md) to include direct links to any cmdlet that gets mentioned there.

File: .docs/Get-VSTeamWorkItemTag.md [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+
+## SYNTAX
+
+## DESCRIPTION
+
+Returns a list of work item tags in a project.
+
+A single tag can be returned by providing the tag name or id.
+
+You must call `Set-VSTeamAccount` before calling this function.
```
Comment (SebastianSchuetze) [inline .docs/Get-VSTeamWorkItemTag.md]: This you don't need to add since this is valid for all cmdlet. There is even an error if you try to do this.

```suggestion
```

File: .docs/Remove-VSTeamWorkItemTag.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+<!-- #include "./synopsis/Get-VSTeamWorkItemTag.md" -->
+
+## SYNTAX
+
+## DESCRIPTION
+
+Removes a work item tag.
+
+You must call `Set-VSTeamAccount` before calling this function.
```
Comment (SebastianSchuetze) [inline .docs/Remove-VSTeamWorkItemTag.md]:
```suggestion
```

File: .docs/Update-VSTeamWorkItemTag.md [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+<!-- #include "./synopsis/Get-VSTeamWorkItemTag.md" -->
+
+## SYNTAX
+
+## DESCRIPTION
+
+Updates a work item tag.
+
+You must call `Set-VSTeamAccount` before calling this function.
```
Comment (SebastianSchuetze) [inline .docs/Update-VSTeamWorkItemTag.md]:
```suggestion
```

File: Source/Public/Get-VSTeamWorkItemTag.ps1 [hunk 4]
```diff
@@ -0,0 +1,38 @@
+function Get-VSTeamWorkItemTag {
+   [CmdletBinding(HelpUri='https://methodsandpractices.github.io/vsteam-docs/docs/modules/vsteam/commands/Get-VSTeamWorkItemTag')]
+   param (
+
+
+      [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
+      [Alias('TagName')]
+      [Alias('TagId')]
+      [string] $TagIdOrName,
```
Comment (SebastianSchuetze) [inline Source/Public/Get-VSTeamWorkItemTag.ps1:7]: Question. API says that in the object ID and name are two different properties. But in the URL it does not care if you use the name or the GUID?

File: Source/Public/Remove-VSTeamWorkItemTag.ps1 [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      [switch] $Force,
+
+      [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
+      [vsteam_lib.ProjectValidateAttribute($false)]
+      [ArgumentCompleter([vsteam_lib.ProjectCompleter])]
+      [string] $ProjectName
+   )
+
+   Process {
+      if ($Force -or $pscmdlet.ShouldProcess($item, "Delete Work Item Tag")) {
```
Comment (SebastianSchuetze) [inline Source/Public/Remove-VSTeamWorkItemTag.ps1]: The message could be a bit clearer. Something like:

"Delete work Item tag from all work items in the project"

File: Source/Public/Update-VSTeamWorkItemTag.ps1 [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+      [switch] $Force,
+
+      [Parameter(Mandatory = $true, ValueFromPipelineByPropertyName = $true)]
+      [vsteam_lib.ProjectValidateAttribute($false)]
+      [ArgumentCompleter([vsteam_lib.ProjectCompleter])]
+      [string] $ProjectName
+   )
+
+   process {
+      if ($Force -or $pscmdlet.ShouldProcess($Id, "Update Work Item Tag")) {
```
Comment (SebastianSchuetze) [inline Source/Public/Update-VSTeamWorkItemTag.ps1]: Same message a bit clearer like with remove.

File: Tests/function/tests/Remove-VSTeamWorkItemTag.Tests.ps1 [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+      Mock Get-VSTeamProject {return Open-SampleFile 'Get-VSTeamProject.json' }            
+   }
+
+   Context 'Remove-VSTeamWorkItemTag' {
+      BeforeAll {
+         $FakeProjectName = 'test'
+         $FakeTagId = '00000000-0000-0000-0000-000000000000'         
+         $FakeTagName = 'myTag'         
+         
+         #TODO
```
Comment (SebastianSchuetze) [inline Tests/function/tests/Remove-VSTeamWorkItemTag.Tests.ps1]: What is missing here?

**Review by SebastianSchuetze — CHANGES_REQUESTED** [review]
Checked your current code.

Since unit tests are failing, then I guess you are not finished yet.

**Comment by SebastianSchuetze** [comment]
@smholvoet any update on your PR?

**Comment by smholvoet** [comment]
@SebastianSchuetze I'll rework this in the next couple of days, thank you for reviewing 👌

Can I go ahead and rework the "related links" section as well? This would impact 172 files (all cmdlets).
Or should I hold off on this until @DarqueWarrior's response clears [this](https://github.com/MethodsAndPractices/vsteam/pull/418#issuecomment-933018716) up? 🤔

> @DarqueWarrior what was the intention to add the standard related texts in every cmdlet? Is this nessecary to mention it everywhere?

**Comment by SebastianSchuetze** [comment]
No it's fine. You can rework. Please just do it in an extra PR.

Thanks for the support. Appreciate it!

**Comment by SebastianSchuetze** [comment]
@smholvoet got any time to finish it? :-)

File: .docs/Get-VSTeamWorkItemTag.md (hunk 1 above)
Comment (smholvoet) [reply .docs/Get-VSTeamWorkItemTag.md]: Removed this line.

File: .docs/Remove-VSTeamWorkItemTag.md (hunk 2 above)
Comment (smholvoet) [reply .docs/Remove-VSTeamWorkItemTag.md]: Removed this line.

File: .docs/Update-VSTeamWorkItemTag.md (hunk 3 above)
Comment (smholvoet) [reply .docs/Update-VSTeamWorkItemTag.md]: Removed this line.

File: Source/Public/Get-VSTeamWorkItemTag.ps1 (hunk 4 above)
Comment (smholvoet) [reply Source/Public/Get-VSTeamWorkItemTag.ps1:7]: Both the Object ID and name should work here, they even called the parameter `tagIdOrName` 🤷‍♂️.

See [Tags - Get](https://docs.microsoft.com/en-us/rest/api/azure/devops/wit/tags/get?view=azure-devops-rest-6.0).

File: Source/Public/Remove-VSTeamWorkItemTag.ps1 (hunk 5 above)
Comment (smholvoet) [reply Source/Public/Remove-VSTeamWorkItemTag.ps1]: Updated the description.

File: Source/Public/Update-VSTeamWorkItemTag.ps1 (hunk 6 above)
Comment (smholvoet) [reply Source/Public/Update-VSTeamWorkItemTag.ps1]: Updated the description.

File: Tests/function/tests/Remove-VSTeamWorkItemTag.Tests.ps1 (hunk 7 above)
Comment (smholvoet) [reply Tests/function/tests/Remove-VSTeamWorkItemTag.Tests.ps1]: Both `Invoke-Restmethod` calls were using an incorrect variable, updated this one.

**Comment by SebastianSchuetze** [comment]
@smholvoet thanks for you work! Please also check the builds. Your unit tests do not seem to go through. Without this we cannot merge.

image

**Review by SebastianSchuetze — CHANGES_REQUESTED** [review]
Just saw the file docs/synopsis/Update-VSTeamWorkItemTag.md

Synopsis is missing there.

You can also run the unit tests locally to check if they work.

https://github.com/MethodsAndPractices/vsteam/blob/trunk/README.md#how-to-build-locally

**Review by SebastianSchuetze — DISMISSED** [review]
Made quite some changes to the unit test. I guess you didn't run them at all locally. :-)

But really thank you for introducing this. I need to check a view more things, then I can merge this.

**Review by SebastianSchuetze — APPROVED** [review]
